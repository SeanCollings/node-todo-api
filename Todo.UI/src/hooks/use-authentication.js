import { useDispatch, useSelector } from 'react-redux';
import { LOCAL_TOKEN } from '../constants';
import { LOGIN_USER_SUCCESS } from '../state/action-types';
import { decodeJWT, getCookie } from '../utils';

const useAuthentication = () => {
  const { user } = useSelector(({ auth }) => auth);
  const dispatch = useDispatch();

  if (!!user) return true;

  const token = getCookie(LOCAL_TOKEN);
  const username = decodeJWT(token);

  if (username?.name) {
    dispatch({
      type: LOGIN_USER_SUCCESS,
      payload: username,
    });
  }

  return !!username;
};

export default useAuthentication;
