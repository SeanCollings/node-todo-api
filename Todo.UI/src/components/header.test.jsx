import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Header from './header';
import { useActions } from '../hooks/use-actions';

jest.mock('../hooks/use-actions');

describe('Header', () => {
  useActions.mockImplementation(() => ({ signOutUser: jest.fn() }));

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render', () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call signoutUser when Log Out button clicked', () => {
    const component = render(<Header />);
    fireEvent.click(component.getByText('Log Out'));
  });
});
