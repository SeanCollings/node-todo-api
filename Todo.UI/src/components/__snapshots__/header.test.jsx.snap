// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div
  className="sc-eCssSg bLRIre"
>
  <div
    className="sc-jSgupP fonzNJ"
  />
  <div
    className="sc-iBPRYJ eBvofg"
  >
    <div
      className="sc-gKsewC flWMrW"
    >
      My Todos
    </div>
    <div
      className="sc-fubCfw ciaMen"
    >
      <button
        className="sc-bdfBwQ sc-gsTCUz udtBZ dmETOR"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
</div>
`;
